### PR TITLE
Makefile go brrrrr

### PR DESCRIPTION
**Pull request by nanaian**

### CODE REVIEW

File: tools/splat.yaml [hunk 1]
```diff
@@ -909,6 +829,7 @@ segments:
     - [0x7E3700, "c"]
     - [0x7e4d00, "bin"]
   - name: world/area_mac/machi/
+    ld_name: machi
```
Comment (ethteck) [inline tools/splat.yaml:832]: I know this is technically a splat thing and I should have caught it there, but I prefer the LD stuff to all be handled behind the scenes as much as possible. What's the reason for this option? What would happen without it?

File: sources.mk [hunk 2]
```diff
@@ -1,35 +1,57 @@
...
-
-OBJECTS = \
-	$(foreach file, $(C_FILES) $(S_FILES) $(BIN_FILES), $(BUILD_DIR)/$(file).o) \
-	$(foreach file, $(COMPRESSED_BIN_FILES), $(BUILD_DIR)/$(file:.bin=.Yay0.o))
+OBJECTS := $(subst BUILD_DIR, $(BUILD_DIR), $(shell grep -E 'BUILD_DIR.+\.o' papermario.ld -o))
+
+DSL_C_FILES := $(shell grep -lrF "SCRIPT" src)
+
+MAPS := \
+	dro_01 dro_02 \
```
Comment (ethteck) [inline sources.mk:6]: It's a bummer to hard-code these. I assume it's for performance reasons? I wish there was a better way to dynamically support areas.

**Review by ethteck — CHANGES_REQUESTED** [review]

File: sources.mk (hunk 2 above)
Comment (nanaian) [reply sources.mk:6]: They have a specific order that I haven't found a pattern to

File: tools/splat.yaml (hunk 1 above)
Comment (nanaian) [reply tools/splat.yaml:832]: It's used for the asset table, which isn't in this PR. It allows us to refer to this as `ROM_machi` etc. (see the generated include/ld_addrs.h).

File: sources.mk (hunk 2 above)
Comment (nanaian) [reply sources.mk:6]: This was in `fs.json` before.

Comment (nanaian) [reply sources.mk:6]: Any mods can just use `find` or something, but we have to adhere to the same order used originally.

File: tools/splat.yaml (hunk 1 above)
Comment (ethteck) [reply tools/splat.yaml:832]: yeah I feel like we should probably make the ld names different, but again my bad for not really catching that in the splat pr. I kinda knew I'd need to see it in practice to have an opinion.

**Pull request merged**